### ISSUE TITLE
Wiki with lib usecases

### ISSUE THREAD

**Issue by confile** [goodow/realtime-json#1]
It would be great if you could publish a wiki with the use cases of this lib. This would help a lot to use it.

**Comment by larrytin** [comment]
Can you explain your problem more specific?

**Comment by drmichaelgorski** [comment]
I do not know how to use you lib.

**Comment by larrytin** [comment]
The start point is com.goodow.realtime.json.Json.parse(jsonString) or com.goodow.realtime.json.Json. createObject().

**Comment by drmichaelgorski** [comment]
How do I use the lib with GWT?

**Comment by larrytin** [comment]
Just include [com.goodow.realtime.json.Json](https://github.com/goodow/realtime-json/blob/master/src/main/resources/com/goodow/realtime/json/Json.gwt.xml) like [this](https://github.com/goodow/realtime-channel/blob/master/src/main/resources/com/goodow/realtime/Platform.gwt.xml), then use Json.parse(jsonString) and Json.createObject()/createArray() as normal.

**Comment by drmichaelgorski** [comment]
Ok do I get only JSONObjects from your lib or can I get Java Class instances like in this lib: https://github.com/nmorel/gwt-jackson

**Comment by larrytin** [comment]
This project doesn't support custom class types, it only support [JsonObject](https://github.com/goodow/realtime-json/blob/master/src/main/java/com/goodow/realtime/json/JsonObject.java), [JsonArray](https://github.com/goodow/realtime-json/blob/master/src/main/java/com/goodow/realtime/json/JsonArray.java)

**Comment by drmichaelgorski** [comment]
Okay thank you. Can you also explain how I use the lib in a J2Objc project? Do I only have to add the lib to the classpath and thats it?

**Comment by larrytin** [comment]
Yes. When you want to translate java sources to objective-c sources, using these  j2objc options: j2objc --mapping [method-mappings.properties](https://github.com/goodow/realtime/blob/master/resources/j2objc/method-mappings.properties) --prefixes [package-prefixes.properties](https://github.com/goodow/realtime/blob/master/resources/j2objc/package-prefixes.properties)

**Comment by drmichaelgorski** [comment]
When I use j2objc I also have to add your JSON lib to the `--classpath`right?

**Comment by larrytin** [comment]
Yes, refer to [this Makefile](https://github.com/goodow/realtime-channel/blob/master/Makefile#L14-L15) to translate [java sources](https://github.com/goodow/realtime-channel/tree/master/src/main/java/) to [objective-c sources](https://github.com/goodow/GDChannel/tree/master/Classes/generated).

**Comment by drmichaelgorski** [comment]
Great I will try it.

**Comment by drmichaelgorski** [comment]
Are you still active develop this library?

**Comment by larrytin** [comment]
Yes, after all, it is very easy to maintain and use.

**Comment by drmichaelgorski** [comment]
Do you plan to provide some kind of object mapping like GWT-Jackson does it?

**Comment by drmichaelgorski** [comment]
Could you please document JacksonUtil.java a little bit more like: 

https://github.com/goodow/realtime-json/blob/master/src/main/java/com/goodow/realtime/json/impl/JacksonUtil.java

**Comment by drmichaelgorski** [comment]
@larrytin I see that JacksonUtil.java uses:  

```
import com.fasterxml.jackson.core.JsonParser;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.SerializationFeature;
```

I did not see these libraries supported by J2Objc: 

```
https://github.com/google/j2objc/blob/master/jre_emul/Makefile
```

Did I miss something here?

**Comment by larrytin** [comment]
It does NOT support object mapping like GWT-Jackson, the JacksonUtil is used for Android or plain JVM only.

**Comment by drmichaelgorski** [comment]
Okay. So could you specify what I can use for GWT/Android/J2Objc that would help a lot.

**Comment by larrytin** [comment]
May refer to the these real projects that using GWT/Android/J2ObjC:
https://github.com/goodow/realtime-channel
https://github.com/goodow/realtime-store

**Comment by drmichaelgorski** [comment]
Also `Json.java`references to `JreJsonFactory`and that to `JacksonUtil`.

So what can I use with J2Objc? Could please give an example how I convert a String into a JSONObject?

Thank you.

**Comment by larrytin** [comment]
[Json.java](https://github.com/goodow/realtime-json/blob/master/src/main/java/com/goodow/realtime/json/Json.java) is overridden by [Json.m](https://github.com/goodow/GDJson/blob/master/Classes/override/com/goodow/realtime/json/Json.m) for J2ObjC, and is overridden by [Json.java](https://github.com/goodow/realtime-json/blob/master/src/main/resources/com/goodow/realtime/json/super/com/goodow/realtime/json/Json.java) for GWT.

**Comment by drmichaelgorski** [comment]
Ah ok so it is just: 

```
JSONObject o = Json.parse(someString);
```

in GWT/Android and J2Objc. Right?

**Comment by larrytin** [comment]
JsonObject obj = Json.parse(someString);
obj.set("aString", "some string");

For iOS, the obj is translated to the standard NSMutableDictionary type.

**Comment by drmichaelgorski** [comment]
Is there a way to get a String from a JsonObject?

**Comment by larrytin** [comment]
Please see:
https://github.com/goodow/realtime-json/blob/master/src/main/java/com/goodow/realtime/json/JsonObject.java#L78-L82

**Comment by drmichaelgorski** [comment]
When I have an instance of some class: 

```
class SomeClass {
 String x; 
 int value;

 public SomeClass(String x, int value) {
   this.x = x;
   this.value = value;
 }
} 
```

When I have an instance like: 

```
SomeClass some = new SomeClass("test", 123);
```

Is there a way to get the JSON String from this class? If so how do I specify which properties will be translated or is it all by default?

**Comment by drmichaelgorski** [comment]
@larrytin Could you please give me a short statement on this issue?

**Comment by larrytin** [comment]
It does NOT support custom class types and object mapping like GWT-Jackson or GSON.
